proof/invariant-abstract/IpcDet_AI.thy

```diff
@@ -1900,6 +1893,10 @@ lemma si_invs'_helper_no_reply:
   apply (fastforce simp: live_def elim!: if_live_then_nonz_capD2)
   done
 
+lemma active_st_tcb_at_awaiting_replyD:
+  "st_tcb_at active t s \<Longrightarrow> st_tcb_at (\<lambda>x. \<not> awaiting_reply x) t s"
+  by (fastforce simp: pred_tcb_at_def obj_at_def split: thread_state.splits)
```

Comment:
I'm generally not a fan of such specific thread_state lemmas. What's the use case, and is it possible to use `pred_tcb_weakenE` instead?

Reply:
Good point, fixed it :+1: